### DISCUSSION TITLE
Cake build is not working

### DISCUSSION THREAD

**Discussion by tkkeyan** [cake-build/cake#3110] [Q&A]
From today morning I am facing this issue in all our application while do the Cake build. Can you please help me on this?

Gitlab-ci.yml:
-----------------
Preview:
    stage: Preview
    script: 'call cake Preview'
    only:
        - Dev2021
    environment:
        name: Preview

Build.cake:
---------------
#addin "nuget:?package=Cake.Npm&version=0.16.0"
#addin "nuget:?package=dejavu.Deploy&version=2019.2.28.1224"
#addin "Cake.Powershell"

Task("Preview")
	.IsDependentOn("Build:Compile")

Error:
-----------
$ call cake Preview
Running build target Preview for environment from build.cake
Preparing to run build script...
Running build script...
Error: Unknown command 'Preview'.
build.cake -target=Preview
  ^^^^^^^ No such command

**Comment by devlead** [comment]
How does the "cake" shell script called look like?

I suspect is has an `-target=` that needs to be adjusted to `--target=`

**Comment by augustoproiete** [comment]
@tkkeyan, with the switch to using a new command line parser, all input arguments must now use `-` when passing a single character argument, and `--` when using an argument that has multiple characters.

As such, if you switch to using: `--target` rather than `-target`, things should hopefully start working for you.

See also this similar question and answer: https://github.com/cake-build/cake/discussions/3100
